src/builtin/tags/for.ts

```diff
@@ -36,7 +36,7 @@ export default {
   },
   render: function * (ctx: Context, emitter: Emitter) {
     const r = this.liquid.renderer
-    let collection = toEnumerable(evalToken(this.collection, ctx))
+    let collection = yield toEnumerable(evalToken(this.collection, ctx))
```

Comment:
Can we yield the value instead of the `toEmumerable()`, that'll keep toEnumerable clean and not aware of promises or/and the generator trick. Like:

```javascript
let collection = toEnumerable(yield evalToken(this.collection, ctx))
```

Reply:
ah sure, one sec. not that familiar with generators tbh

Reply:
Updated! How's that? @harttle

Reply:
View final diff, I have some dummy commits (you can squash / merge?)